src/components/projects/projectId/attributes/attributeId/LLM/LLMPlaygroundModal.tsx

```diff
@@ -0,0 +1,210 @@
+        dummyFilter.push(JSON.stringify({ ORDER_BY: ["RANDOM"], "ORDER_DIRECTION": [generateRandomSeed()] }));
+        searchAndSetWithFilter(dummyFilter);
+    }, [searchAndSetWithFilter]);
+
+    const getByRunningId = useCallback(() => {
+        if (inputRunningIdRef.current.length == 0) return;
+        const dummyFilter = [];
+        dummyFilter.push(JSON.stringify(
+            { RELATION: "NONE", NEGATION: false, TARGET_TABLE: "RECORD", TARGET_COLUMN: "DATA", OPERATOR: "EQUAL", VALUES: ["running_id", Number(inputRunningIdRef.current)] }));
+        searchAndSetWithFilter(dummyFilter);
+    }, [searchAndSetWithFilter]);
+
+    const testConfigurationForRecordId = useCallback(() => {
+        if (!recordDataRef.current || recordDataRef.current?.length == 0) return;
+        setPlaygroundTestRunning(true);
+        const recordIds = recordDataRef.current.map((record) => record.id);
+        runAttributeLlmPlayground(projectId, modalRef.current.attributeId, recordIds, fullLlmConfigRef.current, (res) => {
+            let answer = ""
+            for (const id of recordIds) answer += "Answer: " + (res[id] || "No answer found") + "\n";
+            if (res["logs"]) answer += "\n---\nlogs:\n" + res["logs"].join("\n");
+            setLlmAnswer(answer);
+            setPlaygroundTestRunning(false);
+        });
+    }, []);
+
+    useEffect(() => {
+        if (!recordData || recordData?.length == 0) return;
+        setInputRunningId(recordData[0].running_id);
+    }, [recordData]);
+
+    useEffect(() => {
+        if (modal.open) {
+            const config = jsonCopy(attributeDict[modal.attributeId]?.additionalConfig);
+            config.llmConfig.apiKey = modal.apiKey;
+            setFullLlmConfig(config);
+            get1RandomRecords();
+        } else {
+            setRecordData(null);
+            setLlmAnswer(null);
+        }
+    }, [modal.open]);
+
+
+    const copyToAttributeValues = useCallback(() => {
+        if (!fullLlmConfigRef.current || !modalRef.current) return;
+        const config = { ...fullLlmConfigRef.current };
+        updateAttribute(projectId, modalRef.current.attributeId, (res) => { }, null, null, null, null, null, config);
+    }, []);
+
+    useEffect(() => {
+        if (!attributeDict || !modal?.attributeId) return;
+        const attribute = attributeDict[modal.attributeId];
+        if (attribute.state == AttributeState.USABLE) setAcceptButton({ ...acceptButton, useButton: false });
+        else setAcceptButton({ ...acceptButton, emitFunction: copyToAttributeValues });
+    }, [copyToAttributeValues, attributeDict, modal?.attributeId]);
+
+    const recordKeys = useMemo(() => {
+        if (!attributes) return [];
+        return attributes.filter(a => DISPLAY_STATES.includes(a.state)).map(a => ({ name: a.name, dataType: a.dataType }));
+    }, [attributes])
+
+    return (<Modal modalName={ModalEnum.LLM_PLAYGROUND} acceptButton={acceptButton} modalWidth="ml-10 w-full max-w-[calc(100vw-15rem)]">
```

Comment:
we also implemented some logic for passing the modal width, we can sync on a quick call :)